### ISSUE TITLE
Implement IntoIterator for Archive

### ISSUE THREAD

**Issue by mbrt** [alexcrichton/tar-rs#58]
I found myself wanting to store an `Entries` iterator in a struct. This however means I have to depend on some `Archive` reference that must outlive my struct. Wouldn't it be possible to implement `IntoIterator` for `Archive`? In this way I can store the iterator only, without external dependencies on an `Archive` instance.

**Comment by alexcrichton** [comment]
Sounds like a reasonable extension to me!

**Comment by mbrt** [comment]
@alexcrichton I was trying to see if it's easy to implement, but I found difficult to reuse code between `EntriesFields` and an `IntoIterator` implementation. The first needs a reference to an `Archive`, while the second does not have it. Any clue?

**Comment by alexcrichton** [comment]
Yeah this is probably easiest by having the `next` method of `IntoIter` construct a `EntriesFields` on the fly and then sync the internal fields before/after each call. Does that make sense?

**Comment by mbrt** [comment]
Yes, it does! I'll try...
